app/assets/javascripts/discourse/app/models/topic.js

```diff
@@ -461,7 +462,15 @@ export default class Topic extends RestModel {
   }
 
   set details(value) {
-    this._details = value;
+    const TopicDetails = getOwner(this).factoryFor("model:topic-details").class;
```

Comment:
Ah, so I guess you can't do an `import` for this, as it doesn't actually give you the JS class itself?

Reply:
We could, I was just trying to create a hard dependency from TopicDetails into Topic